pkg/controllers/controllers.go

```diff
@@ -24,13 +25,15 @@ import (
 	"github.com/aws/karpenter-core/pkg/operator/controller"
 	awscontext "github.com/aws/karpenter/pkg/context"
 	"github.com/aws/karpenter/pkg/controllers/interruption"
+	"github.com/aws/karpenter/pkg/controllers/nodetemplate"
 	"github.com/aws/karpenter/pkg/utils/project"
 )
 
 func NewControllers(ctx awscontext.Context, cloudProvider *cloudprovider.CloudProvider) []controller.Controller {
 	logging.FromContext(ctx).With("version", project.Version).Debugf("discovered version")
 	return []controller.Controller{
 		interruption.NewController(ctx.KubeClient, ctx.Clock, ctx.EventRecorder, interruption.NewSQSProvider(sqs.New(ctx.Session)), ctx.UnavailableOfferingsCache),
+		nodetemplate.NewController(ctx.KubeClient, ec2.New(ctx.Session), ctx.SubnetProvider, ctx.SecurityGroupProvider),
```

Comment:
Is it possible that we could pass in the cloud provider here and make the Subnet and SecurityGroup Providers public? It seems like this may increase in size, and it'd be better to just pass in the whole cloud provider. WDYT?

Reply:
I'd lean away from exposing private members of structs to be accessed by other structs. This breaks the typical higherarchical dependency injection pattern and creates interwoven dependencies, which can be difficult to untangle as projects evolve.